src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -631,6 +631,27 @@ public void delete() throws IOException {
         }
     }
 
+    /**
+     * Will archive and this repository as read-only. When a repository is archived, any operation
+     * that can change its state is forbidden. This applies symmetrically if trying to unarchive it.
+     *
+     * <p>When you try to do any operation that modifies a read-only repository, it returns the
+     * response:
+     *
+     * <pre>
+     * org.kohsuke.github.HttpException: {
+     *     "message":"Repository was archived so is read-only.",
+     *     "documentation_url":"https://developer.github.com/v3/repos/#edit"
+     * }
+     * </pre>
+     *
+     * @throws IOException In case of any networking error or error from the server.
+     */
+    public void doArchive() throws IOException {
```

Comment:
Would it make more sense to have `setArchived` instead? That would support archiving and unarchiving.